src/main/java/com/example/springBoot_hibernate_ferrocarriles/controller/ViajeController.java

```diff
@@ -0,0 +1,86 @@
+package com.example.springBoot_hibernate_ferrocarriles.controller;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.springBoot_hibernate_ferrocarriles.service.ViajeService;
+import com.example.springBoot_hibernate_ferrocarriles.service.EquipoDeTraccionService;
+import com.example.springBoot_hibernate_ferrocarriles.model.EquipoDeTraccion;
+import com.example.springBoot_hibernate_ferrocarriles.model.Itinerario;
+import com.example.springBoot_hibernate_ferrocarriles.model.Viaje;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+@RestController
+public class ViajeController {
+	@Autowired
+	ViajeService viajeService;
+	@Autowired
+	EquipoDeTraccionService equipoDeTraccionService;
+
+	public ViajeController() {
+		super();
+	}
+	
+	@GetMapping("/viaje")
+	public List<Viaje> getAllViajes(){
+		return viajeService.getAllViajes();
+	}
+	
+	@GetMapping("/viaje/{idViaje}")
+	public Viaje getViajeById(@PathVariable("idViaje")Long idViaje){
+		return viajeService.getViajeById(idViaje);
+	}
+	
+	@PostMapping("/viaje")
+	public Viaje addViaje(@RequestBody Itinerario itinerario){		
+		int numeroViaje = viajeService.getAllViajes().size()+1;
```

Comment:
`ViajeService` classes should contain this logic, and release the controllers from computing stuff or doing business logic.